src/lib/server/hyrise_communicator.cpp

```diff
@@ -0,0 +1,111 @@
+#include "hyrise_communicator.hpp"
+
+#include "expression/value_expression.hpp"
+#include "hyrise.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "sql/sql_pipeline_builder.hpp"
+#include "sql/sql_translator.hpp"
+
+namespace opossum {
+
+ExecutionInformation HyriseCommunicator::execute_pipeline(const std::string& sql, const bool debug_note) {
+  // A simple query command invalidates unnamed statements
+  if (Hyrise::get().storage_manager.has_prepared_plan("")) Hyrise::get().storage_manager.drop_prepared_plan("");
+
+  auto execution_info = ExecutionInformation();
+  std::unique_ptr<SQLPipeline> sql_pipeline;
+  try {
+    sql_pipeline = std::make_unique<SQLPipeline>(SQLPipelineBuilder{sql}.create_pipeline());
+    sql_pipeline->get_result_table();
+  } catch (const InvalidInputException& exception) {
+    execution_info.error = exception.what();
+    return execution_info;
+  }
+
+  const auto [pipeline_status, result_table] = sql_pipeline->get_result_table();
+  if (pipeline_status == SQLPipelineStatus::Success) {
+    execution_info.result_table = result_table;
+    execution_info.root_operator = sql_pipeline->get_physical_plans().front()->type();
```

Comment:
What is the purpose of that `root_operator`?
Just curious and I cannot search right now (mobile).

Reply:
For the PostgreSQL protocol we need to find out the type of a statement, such as SELECT, INSERT, UPDATE, ...

Reply:
Could you add a short comment to the hpp?